src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java

```diff
@@ -1,27 +1,62 @@
 package frc.robot.hardware.phoenix6.motor;
 
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
 import com.ctre.phoenix6.signals.NeutralModeValue;
 import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.Robot;
 import frc.robot.hardware.interfaces.ControllableMotor;
 import frc.robot.hardware.phoenix6.Phoenix6Device;
 import frc.robot.hardware.interfaces.IRequest;
+import frc.robot.hardware.phoenix6.Phoenix6DeviceID;
+import frc.robot.hardware.phoenix6.motor.simulation.TalonFXSimulation;
+import frc.robot.hardware.mechanisms.MechanismSimulation;
 import frc.robot.hardware.phoenix6.request.Phoenix6Request;
 import frc.utils.alerts.Alert;
 import frc.utils.calibration.sysid.SysIdCalibrator;
 
 public class TalonFXMotor extends Phoenix6Device implements ControllableMotor {
 
+	private static final int APPLY_CONFIG_RETRIES = 5;
+
 	private final TalonFXWrapper motor;
+	private final TalonFXSimulation talonFXSimulation;
 	private final SysIdCalibrator.SysIdConfigInfo sysidConfigInfo;
 
-	public TalonFXMotor(String logPath, TalonFXWrapper motor, SysIdRoutine.Config sysidConfig) {
+	public TalonFXMotor(
+		String logPath,
+		Phoenix6DeviceID deviceID,
+		TalonFXConfiguration configuration,
+		SysIdRoutine.Config sysidConfig,
+		MechanismSimulation simulation
+	) {
 		super(logPath);
-		this.motor = motor;
+		this.motor = Robot.ROBOT_TYPE.isSimulation() ? new TalonFXWrapper(deviceID.ID()) : new TalonFXWrapper(deviceID);
+		if (!motor.applyConfiguration(configuration, APPLY_CONFIG_RETRIES).isOK()) {
+			new Alert(Alert.AlertType.ERROR, getLogPath() + "ConfigurationFailed").report();
+		}
+		this.talonFXSimulation = Robot.ROBOT_TYPE.isSimulation() && simulation != null
```

Comment:
why would you pass a null simulation?

Reply:
If u don’t want to run it, like there is no reason to run simulation code (that takes more run time) when running on real robot for example. The simulation value is being checked before the update if it’s null so it should not make any problem